### PR TITLE
build: update dev npm packages + enable unicorn/recommended

### PR DESCRIPTION
**Pull request by portenez**
**Describe your changes**
Updated dev dependencies to the latest.

For some I ran `npm update`, and for others, I ran `npm install -D {package}@latest`. I ran `npm install -D` for packages that `npm update` wouldn't update (breaking changes)

**Testing performed**
I ran `npm run verify` which runs:

```
> coffeechat@0.0.0 verify
> npm run prettier:verify 
&& npm run eslint 
&& npm run stylelint 
&& npm run build 
&& npm test 
&& npm run storybook:build
```

I also:

- [x] `npm run dev`. 
  - [x] Then check the app runs and no warnings
- [x] `npm run build && npm start`. 
  - [x] Then check the app runs and no warnings
- [x] `npm run storybook`. Check stories render
- [x] `npm run storybook:build && npm run storybook:server`
  - [x] Made sure everything works 

**Additional context**
- I did not install `@storybook/testing-react`, which introduces an eslint warning. The types for that library are returning an `any` that were not there before. I'll update when that is fixed. I might log an issue myself.

### CODE REVIEW

File: jest.config.js [hunk 1]
```diff
@@ -3,4 +3,5 @@ module.exports = {
   testMatch: ['**/__tests__/**/*.spec.[jt]s?(x)'],
   setupFiles: ['<rootDir>/.jest/storyBookTesting.ts'],
   setupFilesAfterEnv: ['<rootDir>/.jest/jestDomSetup.ts'],
+  testEnvironment: 'jsdom',
```
Comment (portenez) [inline jest.config.js:6]: required by the newest `jest` version

File: package.json [hunk 2]
```diff
@@ -74,9 +74,9 @@
     "eslint-plugin-promise": "^5.1.0",
     "eslint-plugin-react": "^7.23.2",
     "eslint-plugin-react-hooks": "^4.2.0",
-    "eslint-plugin-unicorn": "^31.0.0",
-    "jest": "^26.6.3",
-    "prettier": "^2.2.1",
+    "eslint-plugin-unicorn": "^34.0.1",
```
Comment (portenez) [inline package.json:77]: these had breaking changes, and had to use `npm install -D {package}@latest`. I checked, nothing breaks, and a skimmed through the changelog (things look fine).

**Comment by portenez** [comment]
FYI , the bulk of the changes are in the `package-lock.json`. When using `npm update`, `npm` first update the package in place, then updates the lock file.

File: .eslintrc.yml [hunk 3]
```diff
@@ -95,22 +96,33 @@ rules:
...
       '@typescript-eslint/no-var-requires': 0
+      unicorn/prefer-module: 0
   - files:
       - 'scripts/**/*.{ts,tsx,js,jsx}'
     rules:
       no-console: 0
+  - files:
+      - 'next-env.d.ts'
+    rules:
+      unicorn/prevent-abbreviations: 0
```
Comment (Abhilasha3112) [inline .eslintrc.yml]: if we want to disable a single rule for a single file, how about we do it using `eslint-disable`? I feel adding single rule exceptions for individual files might increase the size of eslintrc file and we might want to keep it only for maintaining all generic rules. Not sure though what's the best practice here?

**Review by Abhilasha3112 — COMMENTED** [review]
lgtm except one small doubt

File: .eslintrc.yml (hunk 3 above)
Comment (portenez) [reply .eslintrc.yml]: yeah, makes total sense. Just pushed this change.

**Review by Abhilasha3112 — DISMISSED** [review]

**Comment by portenez** [comment]
@Abhilasha3112 @dakotalillie can you approve again? I squashed a bunch of commits.

**Review by Abhilasha3112 — APPROVED** [review]

**Pull request merged**